duckduckgo/iOS
# Interactive bar animation and scrolling improvements

Task/Issue URL:
https://app.asana.com/0/856498667320406/893432868851070
https://app.asana.com/0/856498667320406/1126337473600500

**Description**:
- Interactive animation of top/bottom bar based on the scrolling progress.
- Tweaks to long press & bottom tap gesture recognizers to minimize impact on scroll responsiveness.

**Steps to test this PR**:

Following scenarios should work:
- Hiding bars when scrolling downwards.
- Revealing bars when scrolling above top of the view.
- Revealing bars when decelerating toward the top.
- Revealing bars when tapping on the bottom area of the screen.
- Revealing bars when tapping on top of the screen.
- Revealing bars on navigation.
- Hiding/showing bars when zooming in/out.
- Hiding disabled when web page viewport is smaller than screen (e.g: ddg lite).
- 'Tiny' swipes should work.
- Long press contextual menu should work.


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
